src/passes/assume.rs

```diff
@@ -0,0 +1,63 @@
+use crate::{
+    core::{self, OrderConstraint, Expr},
+    passes::Pass
+};
+use itertools::Itertools;
+
+/// Add default assumptions to the Filament program
+pub struct Assume;
+
+impl Assume {
+    /// Iterate through the parameter constraints in a signature to see if we can add any default function assumptions
+    fn sig(sig: &core::Signature) -> Vec<core::Command> {
+        sig.param_constraints
+            .iter()
+            .filter_map(|c| match c.inner() {
+                OrderConstraint {
+                    op: core::OrderOp::Eq,
+                    left,
+                    right
+                } => {
+                    match (left, right) {
+                        (
+                            _,
+                            Expr::App { func, arg: right}
+                        ) => Some (func.clone().assume(left, right)),
+                        (
+                            Expr::App { func, arg: left},
+                            _
+                        ) => Some (func.clone().assume(right, left)),
+                        _ => None
+                    }
+                },
+                _ => None
+            })
+            .flatten()
+            .map(core::Command::from)
+            .collect_vec()
+    }
+
+    fn component(comp: core::Component) -> core::Component {
+        let mut pre_cmds = Assume::sig(&comp.sig);
+        pre_cmds.extend(comp.body.into_iter());
+
+        core::Component {
+            body: pre_cmds,
```

Comment:
One possible problem here is that the `assume` statements are only visible in the body and not the signature which means that if the signature needs a property to be true for it to be valid, this will not work. There is no good solution to this yet because signature bindings don't allow you to `assume` things but something to keep in mind for future extension.

Reply:
Created an issue for this at #141 to save it.

Reply:
Perfect!